moshi/moshi/models/lm.py

```diff
@@ -528,22 +534,26 @@ def _init_streaming_state(self, batch_size: int) -> _LMGenState:
         if self.lm_model.fuser is None:
             assert not self.condition_tensors
             condition_sum = None
+            condition_cross = None
         else:
             assert self.condition_tensors is not None
             condition_sum = self.lm_model.fuser.get_sum(self.condition_tensors)
+            condition_cross = self.lm_model.fuser.get_cross(self.condition_tensors)
 
         disable = lm_model.device.type != 'cuda'
         graphed_main = CUDAGraphed(lm_model.forward_text, disable=disable)
         graphed_depth = CUDAGraphed(self.depformer_step, disable=disable)
 
         state = _LMGenState(
             batch_size, lm_model.device, cache, initial, graphed_main, graphed_depth,
-            condition_sum=condition_sum)
+            condition_sum=condition_sum, condition_cross=condition_cross)
 
         if self.cfg_coef != 1.:
             batch_size *= 2
             if state.condition_sum is not None:
-                assert state.condition_sum.shape[0] == batch_size, "CFG requires 2x more conditions."
+                assert state.condition_sum.shape[0] == batch_size, "cfg requires 2x more conditions."
+            if state.condition_cross is not None:
+                assert state.condition_cross.shape[0] == batch_size, "cfg requires 2x more conditions."
```

Comment:
I'm a bit lost here, I would have thought that the condition tensors can have a shape `(1, T, D)`, wouldn't that be allowed here too in case we want the cfg to applies to `condition_sum` but not `condition_cross`?

Reply:
In order to control what the CFG applies to the proper control is at the place where we compute the null condition tensors, e.g by not nullifying some of the attributes before feeding to the condition provider and fuser. Remember sum and cross condition tensors can potentially be aggregate from multiple conditioning, so at this point it is too late to decide to do partial CFG.